File: test-local.js
#!/usr/bin/env node

/**
 * Local Machine Test Script for EuroMillions Analysis App
 * Run with: node test-local.js
 */

import http from 'http';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

console.log('🧪 EuroMillions App - Local Machine Test\n');

// Test configuration
const BASE_URL = 'http://localhost:5000';
const TESTS = [
  { name: 'Health Check', url: '/', expectsHTML: true },
  { name: 'Force Initialize', url: '/api/initialize', expectsJSON: true },
  { name: 'Stats API', url: '/api/stats', expectsJSON: true },
  { name: 'History API', url: '/api/history', expectsJSON: true },
  { name: 'Jackpot API', url: '/api/jackpot', expectsJSON: true },
  { name: 'Next Draw API', url: '/api/next-draw', expectsJSON: true },
  { name: 'Analytics API', url: '/api/analytics/numbers', expectsJSON: true }
];

// Colors for console output
const colors = {
  green: '\x1b[32m',
  red: '\x1b[31m',
  yellow: '\x1b[33m',
  blue: '\x1b[34m',
  reset: '\x1b[0m'
};

function log(message, color = 'reset') {
  console.log(colors[color] + message + colors.reset);
}

// Test HTTP request
function testEndpoint(test) {
  return new Promise((resolve) => {
    const url = BASE_URL + test.url;
    
    const req = http.get(url, (res) => {
      let data = '';
      
      res.on('data', (chunk) => {
        data += chunk;
      });
      
      res.on('end', () => {
        const success = res.statusCode === 200;
        const result = {
          name: test.name,
          url: url,
          status: res.statusCode,
          success: success,
          data: data,
          contentType: res.headers['content-type']
        };
        
        if (success && test.expectsJSON) {
          try {
            result.json = JSON.parse(data);
          } catch (e) {
            result.success = false;
            result.error = 'Invalid JSON response';
          }
        }
        
        resolve(result);
      });
    });
    
    req.on('error', (error) => {
      resolve({
        name: test.name,
        url: url,
        success: false,
        error: error.message
      });
    });
    
    req.setTimeout(5000, () => {
      req.destroy();
      resolve({
        name: test.name,
        url: url,
        success: false,
        error: 'Request timeout'
      });
    });
  });
}

// Check if files exist
function checkFiles() {
  log('📁 Checking project files...', 'blue');
  
  const requiredFiles = [
    'package.json',
    'server/index.ts',
    'client/src/App.tsx',
    'shared/schema.ts',
    '.env'
  ];
  
  const missing = [];
  
  requiredFiles.forEach(file => {
    if (fs.existsSync(file)) {
      log(`✅ ${file}`, 'green');
    } else {
      log(`❌ ${file}`, 'red');
      missing.push(file);
    }
  });
  
  if (missing.length > 0) {
    log(`\n⚠️  Missing files: ${missing.join(', ')}`, 'yellow');
    return false;
  }
  
  return true;
}

// Check environment variables
function checkEnvironment() {
  log('\n🔧 Checking environment...', 'blue');
  
  if (!fs.existsSync('.env')) {
    log('❌ .env file not found', 'red');
    return false;
  }
  
  const envContent = fs.readFileSync('.env', 'utf8');
  const hasDbUrl = envContent.includes('DATABASE_URL');
  
  if (hasDbUrl) {
    log('✅ DATABASE_URL configured', 'green');
  } else {
    log('❌ DATABASE_URL not found in .env', 'red');
    return false;
  }
  
  return true;
}

// Validate API responses
function validateResponse(result) {
  if (!result.success) {
    return false;
  }
  
  switch (result.name) {
    case 'Stats API':
      return result.json && 
             typeof result.json.totalCombinations === 'number' &&
             typeof result.json.drawnCombinations === 'number';
             
    case 'History API':
      return Array.isArray(result.json) && result.json.length > 0;
      
    case 'Jackpot API':
      return result.json && 
             typeof result.json.amountEur === 'number' &&
             typeof result.json.amountZar === 'number';
             
    case 'Next Draw API':
      return result.json && result.json.nextDrawDate;
      
    case 'Analytics API':
      return result.json && 
             Array.isArray(result.json.hotNumbers) &&
             Array.isArray(result.json.coldNumbers);
             
    default:
      return true;
  }
}

// Main test runner
async function runTests() {
  log('🚀 Starting EuroMillions App Tests\n', 'blue');
  
  // Pre-flight checks
  if (!checkFiles() || !checkEnvironment()) {
    log('\n❌ Pre-flight checks failed. Please fix the issues above.', 'red');
    process.exit(1);
  }
  
  log('\n🌐 Testing API endpoints...', 'blue');
  
  const results = [];
  let passed = 0;
  let failed = 0;
  
  for (const test of TESTS) {
    process.stdout.write(`Testing ${test.name}... `);
    
    const result = await testEndpoint(test);
    results.push(result);
    
    if (result.success && validateResponse(result)) {
      log('✅ PASS', 'green');
      passed++;
    } else {
      log('❌ FAIL', 'red');
      if (result.error) {
        log(`   Error: ${result.error}`, 'red');
      } else if (result.status !== 200) {
        log(`   Status: ${result.status}`, 'red');
      }
      failed++;
    }
  }
  
  // Summary
  log('\n📊 Test Results:', 'blue');
  log(`✅ Passed: ${passed}`, 'green');
  log(`❌ Failed: ${failed}`, failed > 0 ? 'red' : 'green');
  
  // Detailed results for passing tests
  log('\n📋 Detailed Results:', 'blue');
  results.forEach(result => {
    if (result.success && result.json) {
      switch (result.name) {
        case 'Force Initialize':
          if (result.json.stats) {
            log(`🔄 Initialization complete: ${result.json.stats.drawnCombinations} draws loaded`);
          }
          break;
        case 'Stats API':
          log(`📈 ${result.json.drawnCombinations} draws out of ${result.json.totalCombinations.toLocaleString()} combinations`);
          break;
        case 'History API':
          log(`📅 ${result.json.length} historical draws loaded`);
          if (result.json[0]) {
            log(`   Latest: ${new Date(result.json[0].drawDate).toDateString()}`);
          }
          break;
        case 'Jackpot API':
          log(`💰 Jackpot: €${result.json.amountEur.toLocaleString()} (R${Math.round(result.json.amountZar).toLocaleString()})`);
          break;
        case 'Next Draw API':
          log(`⏰ Next draw: ${new Date(result.json.nextDrawDate).toLocaleString()}`);
          break;
      }
    }
  });
  
  if (failed === 0) {
    log('\n🎉 All tests passed! Your local setup is working correctly.', 'green');
    log('\n🌟 Open http://localhost:5000 in your browser to use the app.', 'blue');
  } else {
    log('\n⚠️  Some tests failed. Check the server logs and fix the issues.', 'yellow');
  }
  
  log('\n💡 Tip: Make sure the server is running with "npm run dev"', 'blue');
}

// Run the tests
runTests().catch(console.error);

File: test-local.py

#!/usr/bin/env python3
"""
Test script for local EuroMillions database
Python version of test-local.js
"""

import json
import urllib.request
import urllib.error
from datetime import datetime
import sys

BASE_URL = 'http://localhost:5000'

def colored_print(message, color='white'):
    """Print colored text to console"""
    colors = {
        'green': '\033[92m',
        'red': '\033[91m',
        'yellow': '\033[93m',
        'blue': '\033[94m',
        'white': '\033[97m',
        'reset': '\033[0m'
    }
    print(f"{colors.get(color, colors['white'])}{message}{colors['reset']}")

def make_request(url, timeout=10):
    """Make HTTP request and return response"""
    try:
        with urllib.request.urlopen(url, timeout=timeout) as response:
            data = response.read().decode('utf-8')
            return {
                'status': response.status,
                'success': response.status == 200,
                'data': json.loads(data),
                'error': None
            }
    except urllib.error.HTTPError as e:
        return {
            'status': e.code,
            'success': False,
            'data': None,
            'error': f'HTTP {e.code}: {e.reason}'
        }
    except urllib.error.URLError as e:
        return {
            'status': 0,
            'success': False,
            'data': None,
            'error': f'Connection error: {e.reason}'
        }
    except json.JSONDecodeError as e:
        return {
            'status': 0,
            'success': False,
            'data': None,
            'error': f'Invalid JSON: {e}'
        }
    except Exception as e:
        return {
            'status': 0,
            'success': False,
            'data': None,
            'error': f'Unexpected error: {e}'
        }

def run_test(test_name, url, expected_keys=None):
    """Run a single test"""
    colored_print(f"\nTesting {test_name}...", 'blue')
    
    response = make_request(url)
    
    if not response['success']:
        colored_print(f"❌ {test_name} failed", 'red')
        colored_print(f"   Status: {response['status']}", 'red')
        colored_print(f"   Error: {response['error']}", 'red')
        return False
    
    # Check expected keys if provided
    if expected_keys and response['data']:
        missing_keys = [key for key in expected_keys if key not in response['data']]
        if missing_keys:
            colored_print(f"❌ {test_name} failed - missing keys: {missing_keys}", 'red')
            return False
    
    colored_print(f"✅ {test_name} passed", 'green')
    return True

def main():
    colored_print("🧪 EuroMillions Local API Test Suite", 'blue')
    colored_print("=" * 50, 'blue')
    
    tests = [
        ("Server Health", f"{BASE_URL}/api/stats", ['totalCombinations', 'drawnCombinations']),
        ("Draw History", f"{BASE_URL}/api/history", None),
        ("Current Jackpot", f"{BASE_URL}/api/jackpot", ['amountEur', 'amountZar']),
        ("Next Draw Info", f"{BASE_URL}/api/next-draw", ['nextDrawDate', 'timeUntilDraw']),
        ("Number Analytics", f"{BASE_URL}/api/analytics/numbers", ['hotNumbers', 'coldNumbers']),
        ("Star Analytics", f"{BASE_URL}/api/analytics/stars", ['hotStars', 'coldStars']),
        ("Predictions", f"{BASE_URL}/api/predictions", None),
    ]
    
    passed = 0
    failed = 0
    
    for test_name, url, expected_keys in tests:
        if run_test(test_name, url, expected_keys):
            passed += 1
        else:
            failed += 1
    
    # Summary
    colored_print("\n" + "=" * 50, 'blue')
    colored_print("Test Results:", 'blue')
    colored_print(f"✅ Passed: {passed}", 'green')
    colored_print(f"❌ Failed: {failed}", 'red')
    
    if failed == 0:
        colored_print("🎉 All tests passed!", 'green')
        colored_print("Your EuroMillions app is working correctly!", 'green')
    else:
        colored_print("⚠️  Some tests failed. Check the errors above.", 'yellow')
        colored_print("Make sure the server is running with 'python run-local.py'", 'yellow')
    
    colored_print("=" * 50, 'blue')
    
    return 0 if failed == 0 else 1

if __name__ == "__main__":
    sys.exit(main())
